airflow/jobs/scheduler_job.py

```diff
@@ -1194,7 +1195,7 @@ def _schedule_dag_run(
         return callback_to_run
 
     @provide_session
-    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session=None):
+    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session: Session = None) -> None:
```

Comment:
Use `NEW_SESSION` here

Reply:
I've used `NEW_SESSION` across the board, thanks.